### PR TITLE
Chore: separate out projectMachine from Home route

### PR DESCRIPTION
**Pull request by franknoirot**
Prereq for some of the features in #4088. We have not been able to perform any project management commands from within a project up until now, because the project management state machine was embedded within the `/home` route. This makes it available across all routes, so users (or, importantly for #4088, other actors) can perform CRUD operations on projects while inside a project.

Adds 4 E2E tests to check the rename and delete functionality both from home and in the project to be acted upon, since those are the trickier cases. It verifies that we:
1. Navigate home if we delete a project while we are inside it
2. Navigate to the new project name route if we rename a project while we are inside it

### CODE REVIEW

File: src/machines/projectsMachine.ts [hunk 1]
```diff
@@ -30,16 +51,20 @@ export const homeMachine = setup({
   actions: {
     setProjects: assign({
       projects: ({ context, event }) =>
-        'output' in event ? event.output : context.projects,
+        'output' in event && Array.isArray(event.output)
```
Comment (jtran) [inline src/machines/projectsMachine.ts:54]: Use the type guard I made in `utils.ts`.

```suggestion
        'output' in event && isArray(event.output)
```

**Review by jtran — APPROVED** [review]

**Comment by franknoirot** [comment]
Gonna crack open my windows machine tomorrow, I'm so sick of this PR

**Comment by franknoirot** [comment]
Good lord this one was frustrating. But it's green. And now projects can be navigated more effectively.

**Comment by lf94** [comment]
merge it </order-66>

**Pull request merged**